tools/triage/utils/extraction/jira_attachment_querier/jira_remote_archive_querier.py

```diff
@@ -0,0 +1,242 @@
+        self._extract_tar = extract_tar
+        self._extracted = False
+
+        try:
+            self._archive_path = self._get_attachment_from_issue(
+                issue_key=issue_key,
+                attachment_file_name=attachment_file_name,
+                target_directory=self._temporary_download_directory_path,
+            )
+            self._downloaded = True
+
+        except FailedToGetAttachmentInIssue as e:
+            self._logger.exception("failed to get attachment from issue: ", exc_info=str(e))
+            self._downloaded = False
+            return
+
+        # if it is not a tar file, nothing to extract
+        if not LocalNestedArchiveExtractor.is_tar_file(Path(attachment_file_name)):
+            self._is_tar_file = False
+            return
+
+        self._is_tar_file = True
+
+        if not extract_tar:
+            return
+
+        # extracting the tar
+        self.temporary_extracted_archive_path = Path(tempfile.mkdtemp())
+        extraction_path = (
+            self.temporary_extracted_archive_path
+            / LocalNestedArchiveExtractor.remove_tar_and_gzip_extensions_if_any(attachment_file_name)
+        )
+
+        try:
+            LocalNestedArchiveExtractor.recursive_extraction(
+                source=self._archive_path, target=extraction_path, root_tar_path=self._archive_path
+            )
+            self._extracted = True
+
+        except Exception as e:
+            raise FailedToExtractAttachment from e
+
+    def __del__(self):
+        """Destructor. Removes the temporary directories."""
+        try:
+            shutil.rmtree(self._temporary_download_directory_path)
+
+        except AttributeError:
+            pass
+
+        except Exception as e:
+            self._logger.exception("failed to remove download directory", exc_info=str(e))
+
+        try:
+            shutil.rmtree(self.temporary_extracted_archive_path)
+
+        except AttributeError:
+            pass
+
+        except Exception as e:
+            self._logger.exception("failed to remove extraction directory", exc_info=str(e))
+
+    @staticmethod
+    def _download_attachment(attachment, target_directory: Path) -> Path:
+        """Downloads a given attachment to the specified directory.
+
+        Args:
+            attachment: The attachment object to download.
+            target_directory (Path): The directory where the attachment should be saved.
+
+        Returns:
+            Path to the downloaded attachment.
+
+        Raises:
+            Exception: If the download fails.
+        """
+        attachment_content = attachment.get()
+        new_path = target_directory / str(attachment.filename)
+
+        try:
+            with open(new_path, "wb") as file:
+                file.write(attachment_content)
+
+            return new_path
+
+        except Exception as e:
+            raise Exception(f"Failed to download the attachment to {new_path})") from e
```

Comment:
Both catching and raising ``Exception`` is wrong, as a general rule
In this specific case I would just remove it, us people know to read tracebacks and find where errors are coming from :)

Reply:
Also defining custom exceptions. Unless for the use-case of a library that wants its users to differentiate between different problems (e.g., ``requests`` that can return different errors if it's connection error, TLS problem, or something else), there's no need to have this

Let's apply those throughout this PR, we have a chance to remove a lot of code this way and to make it much easier to read and grasp

Reply:
@osherdp  We want to let the script run on all issues even if it failed to generate signatures for one of them, so we must catch some exceptions. Do you think we should raise all exceptions in the end ?

Reply:
I think we already do that on a per-issue basis (letting it run for the rest of the issues even if we encounter an error for one of them)

Reply:
I fixed the PR. I tried to avoid raising/catching/custom exceptions as much as possible, but I also wanted to make as less changes as possible to the code in order to avoid unnecessary issues, so I left the exception classes that was already there as some of the current code depends on it. With that being said, I also raises `FileNotFoundError` if the `jira` attachment download fails, in order to fit the behavior of `nestedarchive` in the case of fail